tensorflow/lite/micro/examples/experimental/image_recognition/image_recognition_test.cc

```diff
@@ -0,0 +1,95 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "tensorflow/lite/micro/examples/experimental/image_recognition/image_recognition_model.h"
+#include "tensorflow/lite/micro/examples/experimental/image_recognition/util.h"
+#include "tensorflow/lite/micro/kernels/all_ops_resolver.h"
+#include "tensorflow/lite/micro/micro_error_reporter.h"
+#include "tensorflow/lite/micro/micro_interpreter.h"
+#include "tensorflow/lite/micro/testing/micro_test.h"
+#include "tensorflow/lite/schema/schema_generated.h"
+#include "tensorflow/lite/version.h"
+
+#include "tensorflow/lite/micro/examples/experimental/image_recognition/first_10_cifar_images.h"
+
+#define IMAGE_BYTES 3072
+#define LABEL_BYTES 1
+#define ENTRY_BYTES (IMAGE_BYTES + LABEL_BYTES)
+
+TF_LITE_MICRO_TESTS_BEGIN
+
+TF_LITE_MICRO_TEST(TestImageRecognitionInvoke) {
+  tflite::MicroErrorReporter micro_error_reporter;
+  tflite::ErrorReporter* error_reporter = &micro_error_reporter;
+
+  const tflite::Model* model = ::tflite::GetModel(image_recognition_model_data);
+  if (model->version() != TFLITE_SCHEMA_VERSION) {
+    error_reporter->Report(
+        "Model provided is schema version %d not equal "
+        "to supported version %d.\n",
+        model->version(), TFLITE_SCHEMA_VERSION);
+  }
+
+  tflite::ops::micro::AllOpsResolver resolver;
+
+  const int tensor_arena_size = 63 * 1024;
+  uint8_t tensor_arena[tensor_arena_size];
+  tflite::MicroInterpreter interpreter(model, resolver, tensor_arena,
+                                       tensor_arena_size, error_reporter);
+  interpreter.AllocateTensors();
+
+  TfLiteTensor* input = interpreter.input(0);
+  TF_LITE_MICRO_EXPECT_NE(nullptr, input);
+  TF_LITE_MICRO_EXPECT_EQ(4, input->dims->size);
+  TF_LITE_MICRO_EXPECT_EQ(1, input->dims->data[0]);
+  TF_LITE_MICRO_EXPECT_EQ(32, input->dims->data[1]);
+  TF_LITE_MICRO_EXPECT_EQ(32, input->dims->data[2]);
+  TF_LITE_MICRO_EXPECT_EQ(3, input->dims->data[3]);
+  TF_LITE_MICRO_EXPECT_EQ(kTfLiteUInt8, input->type);
+
+  int num_correct = 0;
+  int num_images = 10;
+  for (int image_num = 0; image_num < num_images; image_num++) {
+    memset(input->data.uint8, 0, input->bytes);
+
+    uint8_t correct_label = 0;
+
+    correct_label =
+        tensorflow_lite_micro_tools_make_downloads_cifar10_test_batch_bin
+            [image_num * ENTRY_BYTES];
+    memcpy(input->data.uint8,
+           &tensorflow_lite_micro_tools_make_downloads_cifar10_test_batch_bin
+               [image_num * ENTRY_BYTES + LABEL_BYTES],
+           IMAGE_BYTES);
+    reshape_cifar_image(input->data.uint8, IMAGE_BYTES);
+
+    TfLiteStatus invoke_status = interpreter.Invoke();
+    TF_LITE_MICRO_EXPECT_EQ(kTfLiteOk, invoke_status);
+    if (invoke_status != kTfLiteOk) {
+      error_reporter->Report("Invoke failed\n");
```

Comment:
`TF_LITE_REPORT_ERROR should be used instead of error_reporter, so log strings can be removed to save space if needed.`